int32 nb_obj
---